### PR TITLE
build: upgrade rust toolchain to fix ci issues on Windows

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
- https://github.com/GreptimeTeam/greptimedb/issues/3902
- This is likely related to https://github.com/rust-lang/rust/issues/123999

## What's changed and what's your intention?
https://github.com/GreptimeTeam/greptimedb/actions/runs/9009678033/job/24754350747
```
     Running `target\debug\sqlness-runner.exe`
error: process didn't exit successfully: `target\debug\sqlness-runner.exe` (exit code: 0xc0000142, STATUS_DLL_INIT_FAILED)
Error: Process completed with exit code 1.
```

Upgrading the rust toolchain should fix this.
- https://github.com/GreptimeTeam/greptimedb/actions/runs/9043603077
- https://github.com/GreptimeTeam/greptimedb/actions/runs/9043764916

### TODO
We also need to release a new dev-builder after this PR: https://github.com/GreptimeTeam/greptimedb/actions/workflows/release-dev-builder-images.yaml
- [ ] Release dev-builder

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/common/runtime/src/repeated_task.rs [hunk 1]
```diff
@@ -204,7 +204,7 @@ mod tests {
         tokio::time::sleep(Duration::from_millis(550)).await;
         task.stop().await.unwrap();
 
-        assert_eq!(n.load(Ordering::Relaxed), 5);
+        assert!(n.load(Ordering::Relaxed) >= 3);
```
Comment (evenyag) [inline src/common/runtime/src/repeated_task.rs:207]: This test is unstable so I relax the assertion.

**Review by tisonkun — APPROVED** [review]
This should fix https://github.com/GreptimeTeam/greptimedb/issues/3902?

**Comment by evenyag** [comment]
> This should fix #3902?

I think so.

File: src/common/runtime/src/repeated_task.rs (hunk 1 above)
Comment (killme2008) [reply src/common/runtime/src/repeated_task.rs:207]: Sleep for more time may work.

**Review by waynexia — APPROVED** [review]

File: src/common/runtime/src/repeated_task.rs (hunk 1 above)
Comment (evenyag) [reply src/common/runtime/src/repeated_task.rs:207]: If we sleep more time, then we might need to increase the expected value passed to `assert_eq` and encounter the same issue. We can still ensure the repeated task executes repeatedly by checking `n >= 3`.

**Comment by evenyag** [comment]
The dev-builder action 💨 
https://github.com/GreptimeTeam/greptimedb/actions/runs/9075054114

**Pull request merged**